python/pyspark/sql/readwriter.py

```diff
@@ -180,6 +180,9 @@ def json(self, path, schema=None, primitivesAsString=None, prefersDecimal=None,
         :param path: string represents path to the JSON dataset,
                      or RDD of Strings storing JSON objects.
         :param schema: an optional :class:`StructType` for the input schema.
+        :param samplingRatio: sets the ratio for sampling and reading the input data to infer
```

Comment:
it was actually intentional that samplingRatio was undocumented, because regardless the value, Spark still needs to read all the data so this might as well be 1 all the time.